diff --git a/Makefile.inc b/Makefile.inc
--- a/Makefile.inc	2015-10-03 01:35:42.000000000 +0200
+++ b/Makefile.inc	2015-10-02 23:53:18.000000000 +0200
@@ -133,14 +133,14 @@
 WARNINGS +=	-Winline
 WARNINGS +=	-Wdisabled-optimization
 # WARNINGS +=	-Werror
-WARNINGS +=	-finline-functions-called-once
+#WARNINGS +=	-finline-functions-called-once
 WARNINGS +=	-funit-at-a-time
-WARNINGS +=	-fearly-inlining
+#WARNINGS +=	-fearly-inlining
 ifeq ($(DEBUG),0)
 WARNINGS +=	-fomit-frame-pointer
 endif
 # we have small inline functions in src/lq_route.h which should always be inlined
-WARNINGS +=	-finline-limit=350
+#WARNINGS +=	-finline-limit=350
 # These tell gcc to put each function and global variable in a separate section.
 # The linker can than remove all unreferenced section. But in the olsrd binary
 # unused doesn't imply unused at all since the function may be used by plugins,
