diff --git a/Makefile b/Makefile
index 72e92d4..7f975bc 100644
--- a/Makefile
+++ b/Makefile
@@ -219,15 +219,15 @@ rpm:

 # This is quite ugly but at least it works
 ifeq ($(OS),linux)
-SUBDIRS := arprefresh bmf dot_draw dyn_gw dyn_gw_plain httpinfo jsoninfo mdns mini nameservice p2pd pgraph pud quagga secure sgwdynspeed txtinfo watchdog
+SUBDIRS := arprefresh bmf dot_draw dyn_gw dyn_gw_plain jsoninfo mdns mini nameservice p2pd pgraph pud quagga secure sgwdynspeed txtinfo watchdog olsrd_qaul
 else
 ifeq ($(OS),win32)
-SUBDIRS := dot_draw httpinfo jsoninfo mini pgraph secure txtinfo
+SUBDIRS := dot_draw httpinfo jsoninfo mini pgraph secure txtinfo olsrd_qaul
 else
 ifeq ($(OS),android)
-SUBDIRS := arprefresh bmf dot_draw dyn_gw dyn_gw_plain httpinfo jsoninfo mdns mini nameservice p2pd pgraph pud secure sgwdynspeed txtinfo watchdog
+SUBDIRS := arprefresh bmf dot_draw dyn_gw dyn_gw_plain httpinfo jsoninfo mdns mini nameservice p2pd pgraph pud secure sgwdynspeed txtinfo watchdog olsrd_qaul
 else
-SUBDIRS := dot_draw httpinfo jsoninfo mini nameservice pgraph secure txtinfo watchdog
+SUBDIRS := dot_draw httpinfo jsoninfo mini nameservice pgraph secure txtinfo watchdog dyn_gw olsrd_qaul
 endif
 endif
 endif
