diff --git a/build.mak.in.orig b/build.mak.in
index 8a986d7..bdcbf4a 100644
--- a/build.mak.in.orig
+++ b/build.mak.in
@@ -7,7 +7,7 @@ export MACHINE_NAME := auto
 export OS_NAME := auto
 export HOST_NAME := unix
 export CC_NAME := gcc
-export TARGET_NAME := @target@
+export TARGET_NAME := qaul
 export CROSS_COMPILE := @ac_cross_compile@
 export LINUX_POLL := @ac_linux_poll@ 
 export SHLIB_SUFFIX := @ac_shlib_suffix@
