diff --git a/socat_buildscript_for_android.sh b/socat_buildscript_for_android.sh.orig
index 464af43..236e801 100755
--- a/socat_buildscript_for_android.sh
+++ b/socat_buildscript_for_android.sh.orig
@@ -1,7 +1,7 @@
-#!/bin/sh
+#!/bin/bash
 
 # Customize these parameters according to your environment
-ANDROID_NDK="${HOME}/bin/android-ndk-r6b"
+#ANDROID_NDK="${HOME}/bin/android-ndk-r6b"
 
 # Check for parameters
 if [ ! -d "${ANDROID_NDK}" ]; then
@@ -16,7 +16,7 @@ if [ ! -e "${ANDROID_NDK}/build/tools/make-standalone-toolchain.sh" ]; then
 fi
 
 # Extract the Android toolchain from NDK
-ANDROID_PLATFORM="android-3"
+#ANDROID_PLATFORM="android-3"
 ROOT="`pwd`"
 OUT="${ROOT}/out"
 ${ANDROID_NDK}/build/tools/make-standalone-toolchain.sh \
diff --git a/configure.in.orig b/configure.in
index 549f929..d389dae 100644
--- a/configure.in.orig
+++ b/configure.in
@@ -41,7 +41,7 @@ AC_CHECK_PROG(AR, ar, ar, gar)
 # fail
 AC_LANG_COMPILER_REQUIRE()
 
-if test "$CC" = "gcc"; then
+if test "$GCC" = "yes"; then
    CFLAGS="$CFLAGS -D_GNU_SOURCE -Wall -Wno-parentheses"
    ERRONWARN="-Werror -O0"
 elif test "$CC" = "clang"; then
