diff --git a/socat_buildscript_for_android.sh b/socat_buildscript_for_android.sh
index 464af43..b327d41 100755
--- a/socat_buildscript_for_android.sh
+++ b/socat_buildscript_for_android.sh
@@ -1,7 +1,7 @@
-#!/bin/sh
+#!/bin/bash
 
 # Customize these parameters according to your environment
-ANDROID_NDK="${HOME}/bin/android-ndk-r6b"
+#ANDROID_NDK="${HOME}/bin/android-ndk-r6b"
 
 # Check for parameters
 if [ ! -d "${ANDROID_NDK}" ]; then
@@ -16,12 +16,13 @@ if [ ! -e "${ANDROID_NDK}/build/tools/make-standalone-toolchain.sh" ]; then
 fi
 
 # Extract the Android toolchain from NDK
-ANDROID_PLATFORM="android-3"
+#ANDROID_PLATFORM="android-3"
 ROOT="`pwd`"
 OUT="${ROOT}/out"
 ${ANDROID_NDK}/build/tools/make-standalone-toolchain.sh \
-  --ndk-dir="${ANDROID_NDK}" \
+ \
  --platform="${ANDROID_PLATFORM}" \
+ --toolchain="${ANDROID_TOOLCHAIN}" \
  --install-dir="${OUT}/toolchain" \
  || exit 1
 # Remove resolv.h because it is quite unusable as is
