diff --git a/Android.mk b/Android.mk
new file mode 100644
index 0000000..a8708e5
--- /dev/null
+++ b/Android.mk
@@ -0,0 +1,24 @@
+LOCAL_PATH:= $(call my-dir)
+################## build iwlib ###################
+include $(CLEAR_VARS)
+LOCAL_SRC_FILES := iwlib.c
+LOCAL_CFLAGS += -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Winline -MMD -fPIC
+LOCAL_MODULE:= libiw
+LOCAL_STATIC_LIBRARIES := libcutils libc libm
+include $(BUILD_STATIC_LIBRARY)
+################## build iwconfig ###################
+include $(CLEAR_VARS)
+LOCAL_SRC_FILES := iwconfig.c
+LOCAL_CFLAGS += -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Winline -MMD -fPIC
+LOCAL_MODULE:= iwconfig
+LOCAL_STATIC_LIBRARIES := libcutils libc libm libiw
+LOCAL_MODULE_PATH := $(TARGET_OUT_OPTIONAL_EXECUTABLES) # install to system/xbin
+include $(BUILD_EXECUTABLE)
+################## build iwlist ###################
+include $(CLEAR_VARS)
+LOCAL_SRC_FILES := iwlist.c
+LOCAL_CFLAGS += -Wstrict-prototypes -Wmissing-prototypes -Wshadow -Wpointer-arith -Wcast-qual -Winline -MMD -fPIC
+LOCAL_MODULE:= iwlist
+LOCAL_STATIC_LIBRARIES := libcutils libc libm libiw
+LOCAL_MODULE_PATH := $(TARGET_OUT_OPTIONAL_EXECUTABLES) # install to system/xbin
+include $(BUILD_EXECUTABLE)
diff --git a/wireless.22.h b/wireless.h
index 9a885db..d379ec6 100644
--- a/wireless.22.h
+++ b/wireless.h
@@ -1136,4 +1136,7 @@ struct iw_event
 #define IW_EV_QUAL_PK_LEN      (IW_EV_LCP_PK_LEN + sizeof(struct iw_quality))
 #define IW_EV_POINT_PK_LEN     (IW_EV_LCP_LEN + 4)
 
+#undef IW_EV_LCP_PK_LEN
+#undef IW_EV_POINT_PK_LEN
+
 #endif /* _LINUX_WIRELESS_H */
